src/parser/jobs/mch/modules/Wildfire.tsx

```diff
@@ -0,0 +1,204 @@
+import {RotationTable} from 'components/ui/RotationTable'
+import {Event, Events} from 'event'
+import {Analyser} from 'parser/core/Analyser'
+import {EventHook} from 'parser/core/Dispatcher'
+import {filter} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {History, HistoryEntry} from 'parser/core/modules/ActionWindow/History'
+import {Data} from 'parser/core/modules/Data'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import {Timeline} from 'parser/core/modules/Timeline'
+import React, {Fragment} from 'react'
+import {Message} from 'semantic-ui-react'
+
+// We always want 6 GCDs in WF
+const EXPECTED_GCDS = 6
+
+const SEVERITIES = {
+	BAD_WILDFIRE: {
+		1: SEVERITY.MINOR,
+		2: SEVERITY.MEDIUM,
+		3: SEVERITY.MAJOR,
+	},
+	FIZZLED_WILDFIRE: {
+		1: SEVERITY.MEDIUM,
+		2: SEVERITY.MAJOR,
+	},
+}
+
+interface WildfireWindow {
+	actionEvents: Array<Events['action']>
+	stacks: number
+	damage?: number
+}
+
+export class Wildfire extends Analyser {
+	static override handle = 'wildfire'
+	static override title = t('mch.wildfire.title')`Wildfire`
+
+	@dependency private data!: Data
+	@dependency private suggestions!: Suggestions
+	@dependency private timeline!: Timeline
+
+	private history = new History<WildfireWindow>(
+		() => ({
+			actionEvents: [],
+			stacks: 0,
+		})
+	)
+
+	private actionHook?: EventHook<Events['action']>
+
+	private actionFilter = filter<Event>()
+		.source(this.parser.actor.id)
+		.type('action')
+
+	override initialise() {
+		this.addEventHook(filter<Event>()
+			.source(this.parser.actor.id)
+			.type('statusApply')
+			.status(this.data.statuses.WILDFIRE.id)
+		, this.onApply)
+
+		this.addEventHook(filter<Event>()
+			.source(this.parser.actor.id)
+			.type('statusRemove')
+			.status(this.data.statuses.WILDFIRE.id)
+		, this.onRemove)
+
+		this.addEventHook(filter<Event>()
+			.source(this.parser.actor.id)
+			.type('damage')
+			.cause(this.data.matchCauseStatusId([this.data.statuses.WILDFIRE.id]))
+		, this.onDamage)
+
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onApply(event: Events['statusApply']) {
+		if (this.history.getCurrent() != null) {
+			this.history.doIfOpen(current => current.stacks++)
+			return
+		}
+
+		// First application of WF, start a new window
+		this.history.openNew(event.timestamp)
+
+		if (this.actionHook == null) {
+			this.actionHook = this.addEventHook(this.actionFilter, this.onAction)
+		}
+	}
+
+	private onDamage(event: Events['damage']) {
+		this.history.doIfOpen(current => current.damage = event.targets[0].amount)
+	}
+
+	private onRemove(event: Events['statusRemove']) {
+		this.history.closeCurrent(event.timestamp)
+
+		if (this.actionHook != null) {
+			this.removeEventHook(this.actionHook)
+			this.actionHook = undefined
+		}
+	}
+
+	private onAction(event: Events['action']) {
+		this.history.doIfOpen(current => current.actionEvents.push(event))
```

Comment:
Is there any filtering going on to  make sure that the target of your actions is the one afflicted with WF in multi-target situations? Looking through this, it seems like it's all based purely on the WF buff on the actual player. Same goes for the compound damage tracking. Basically what I'm wondering is if this will give weird results in a 2-target scenario (e.g. TEA) if you were to apply WF to one target and then unload on the other.

Reply:
The logic is counting WF debuff stacks themselves, so the GCD count should always be correct. Somehow I missed that this will throw off the "ghosting" logic though, and I can't think of a good way to resolve it so I'm opting to just leave it out. I updated the module description to hopefully make it clear why the GCD count may be lower than the # of GCDs shown in the rotation column.